avocado/utils/network.py

```diff
@@ -45,36 +46,45 @@ def is_port_free(port, address):
     return free
 
 
-def find_free_port(start_port, end_port, address="localhost"):
+def find_free_port(start_port, end_port, address="localhost", sequent=True):
     """
     Return a host free port in the range [start_port, end_port].
 
-    :param start_port: First port that will be checked.
-    :param end_port: Port immediately after the last one that will be checked.
+    :param start_port: header of candidate port range
+    :param end_port: ender of candidate port range
+    :param sequent: Find port sequently, random order if it's False
     :param address: Socket address to bind or connect
     """
-    for i in range(start_port, end_port):
+    port_range = range(start_port, end_port)
```

Comment:
We're pursuing Python 3 compatibility, so this needs an import from six:

```python
six.moves import xrange as range
```

And it needs to be treated as a iterator.  Which means that it needs to be converted to a list before being shuffled.

Reply:
Got it.